src/cockpit/channel.py

```diff
@@ -276,14 +276,14 @@ def do_resume_send(self) -> None:
         """Called to indicate that the channel may start sending again."""
         # change to `raise NotImplementedError` after everyone implements it
 
-    def send_message(self, **kwargs):
-        self.send_channel_message(self.channel, **kwargs)
+    def send_message(self, _msg: 'JsonObject | None' = None, **kwargs: JsonDocument) -> None:
+        self.send_channel_data(self.channel, print_object(_msg, kwargs))
```

Comment:
This relates to my [previous question/confusion](https://github.com/cockpit-project/cockpit/pull/19637#discussion_r1401546089). AFAICS all keys in `send_message()` calls are completely under our control, and we don't actually have any keys containing `-` or `_`. I also don't see any binary data. In other words, we so far have treated this as "pure JSON" protocol without anything funny. Why do we even need this extra msg vs. kwargs split and print_object() magic, as opposed to a plain `json.dumps(kwargs)`?

Reply:
From chat: This can't be spotted with `git grep send_message src/cockpit/`. One case is `max_read_size` which is passed from JS, and forwarding it to a peer via `**kwargs` then reformats the key as `max-read-size`, which doesn't work.

Reply:
This is an existing bug which is fixed by this work.  There are a couple of other similar cases — one in metrics, I think.  Another point is that we wanted to treat `None` values on kwargs differently at some point but were never able to do it because we need to faithfully pass through None from other senders (I tried changing this and it breaks things).